data/calculated_parser/functions.py

```diff
@@ -69,11 +70,293 @@ def sspeed(depth, latitude, temperature, salinity):
         press = [seawater.pres(d, latitude) for d in depth]
     except TypeError:
         press = seawater.pres(depth, latitude)
-
+    
     speed = seawater.svel(salinity, temperature, press)
     return np.array(speed)
 
 
+def sspeedmin(depth, lat, lon, temperature, salinity):
+    """
+    Finds the global minimum of the speed of sound
+
+    Parameters:
+    sspeed: Speed of Sound
+    latitude: The latitude(s) in degrees North
+    """
+    
+    speed = sspeed(depth, lat, temperature, salinity)
+    speed = speed.transpose()
+    
+    for x in range(speed.shape[0]):
+        for y in range(speed.shape[1]):
+            min_val = np.nanmin(speed[x,y])
+            idx = np.where(speed[x,y] == min_val)
+            if (np.isnan(min_val)):
+                speed[x,y] = np.nan
+            elif idx[0].shape[0] > 1:
+                idx = idx[0][0]
+
+                speed[x,y] = depth.values[idx]
+            else:
+                speed[x,y] = depth.values[idx]
+            #speed[x][y] = depth[np.where(speed[x][y] == np.nanmin(speed[x][y]))]  #np.nanmin(speed[x][y])
+    
+    speed = speed.transpose()
```

Comment:
I'm wondering why do we need to perform a transpose twice?

Reply:
This was part of my initial attempts at making it work. I'm not sure what the transpose does at all... I'll investigate on monday to see if it's needed

Reply:
Removed